### PR TITLE
INC-1165 Performance improvements for Case  Note type count

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/CaseNoteService.java [hunk 1]
```diff
@@ -236,13 +238,40 @@ public List<CaseNoteUsageByBookingId> getCaseNoteUsageByBookingId(final String t
...
+            .map(summary ->
+            {
+                final var key = summary.getKey();
+                final var value = summary.getValue();
+                return new CaseNoteUsageByBookingId(
+                    key.bookingId,
+                    key.type,
+                    key.subType,
+                    value.size(),
+                    value.stream().max(Comparator.comparing(OffenderCaseNote::getOccurrenceDateTime)).orElseThrow().getOccurrenceDateTime()
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/CaseNoteService.java:263]: doesn't look like this field is used in incentives - https://github.com/ministryofjustice/hmpps-incentives-api/blob/ad748b2342caf47643f94deced8f82ae6392b0fb/src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonApiService.kt#L75

Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/service/CaseNoteService.java:263]: Its not - This DTO is overloaded so I could create a new one without the field

File: src/main/java/uk/gov/justice/hmpps/prison/service/CaseNoteService.java [hunk 2]
```diff
@@ -236,13 +238,40 @@ public List<CaseNoteUsageByBookingId> getCaseNoteUsageByBookingId(final String t
...
+                );
+            })
+            .sorted(Comparator
```
[the remark this answers is not in the document]
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/service/CaseNoteService.java:266]: No don't need to sort

**Comment by Mjwillis** [comment]
@petergphillips I'll merge and re-submit change

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**